src/humanloop/eval_utils.py

```diff
@@ -0,0 +1,543 @@
+YELLOW = "\033[93m"
+CYAN = "\033[96m"
+GREEN = "\033[92m"
+RED = "\033[91m"
+RESET = "\033[0m"
+
+
+class Identifiers(TypedDict, total=False):
+    """Common identifiers for the objects required to run an Evaluation."""
+    id: NotRequired[str]
+    """The ID of the File on Humanloop."""
+    path: NotRequired[str]
+    """The path of the File on Humanloop."""
+
+
+class File(Identifiers):
+    """A File on Humanloop (Flow, Prompt, Tool, Evaluator)."""
+    type: NotRequired[Literal["flow", "prompt", "tool", "evaluator"]]
+    """The type of File this pipeline relates to on Humanloop."""
+    version: NotRequired[Version]
+    """The contents uniquely define the version of the File on Humanloop"""
+
+
+class RunDataset(Identifiers):
+    datapoints: Sequence[DatapointDict]
+    """The datapoints to map your pipeline over to produce the outputs required by the evaluation."""
+    action: NotRequired[UpdateDatasetAction]
+    """How to update the Dataset given the provided Datapoints; 
+    `set` replaces the existing Datapoints and `add` appends to the existing Datapoints."""
+
+
+class RunEvaluator(Identifiers):
+    """A to provide judgments for this Evaluation."""
+    args_type: NotRequired[EvaluatorArgumentsType]
+    """The type of arguments the Evaluator expects - only required for local Evaluators."""
+    return_type: NotRequired[EvaluatorReturnTypeEnum]
+    """The type of return value the Evaluator produces - only required for local Evaluators."""
+    callable: NotRequired[Callable]
+    """The function to run on the logs to produce the judgment - only required for local Evaluators."""
+    custom_logger: NotRequired[Callable]
+    """optional function that logs the output judgment from your Evaluator to Humanloop, if provided, it will be called as follows:
+    ```
+    judgment = callable(log_dict)
+    log = custom_logger(client, judgmemt)
+    ```
+    Inside the custom_logger, you can use the Humanloop `client` to log the judgment to Humanloop.
+    If not provided your pipline must return a single string.
+    """
+    threshold: NotRequired[float]
+    """The threshold to check the Evaluator against."""
+
+
+class EvaluatorCheck(BaseModel):
+    """Summary data for an Evaluator check."""
+    path: str
+    """The path of the Evaluator used in the check."""
+    improvement_check: bool
+    """Whether the latest version of your pipeline has improved across for a specific Evaluator."""
+    score: float
+    """The score of the latest version of your pipeline for a specific Evaluator."""
+    delta: float
+    """The change in score since the previous version of your pipeline for a specific Evaluator."""
+    threshold: float | None
+    """The threshold to check the Evaluator against."""
+    threshold_check: bool | None
+    """Whether the latest version has an average Evaluator result above a threshold."""
+
+# TODO: Explore restarting runs, but without having to regenerate logs
+
+
+def _run_eval(
```

Comment:
_run_eval is private and not used in this module

Reply:
its a minor abuse of notation; but we want users to only use this via the client evaluations.run (see client.py). So indicating it is private here i think is helpful in that regard

Reply:
Per irl discussion, add a comment explaining the use of the function and close this thread